speculation-rules.bs

```diff
@@ -261,6 +362,19 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
   * |prefetchRecord|'s [=prefetch record/anonymization policy=] equals |prefetchCandidate|'s [=prefetch candidate/anonymization policy=]
 </div>
 
+<div algorithm>
+  To <dfn>find matching links</dfn> given a {{Document}} |document| and [=document rule predicate=] |predicate|:
+
+  1. Let |links| be an empty [=list=].
+  1. For each [=shadow-including descendant=] |descendant| of |document|, in [=shadow-including tree order=]:
+    1. If |descendant| is not an <{a}> with an <{a/href}> attribute or <{area}> element with an <{area/href}> attribute, [=iteration/continue=].
+
+        <div class="note">This corresponds to the elements which match the <a dfn-type="selector">:link</a> and <a dfn-type="selector">:visited</a> [=pseudo-classes=], or which appear in the {{Document/links}} collection.</div>
```

Comment:
Can we use `:any-link` instead?

Reply:
We could. My reason for this choice is this corresponds more directly to [how it's defined in HTML](https://html.spec.whatwg.org/multipage/semantics-other.html#selector-link) and because I think `:any-link` is a little less familiar to most readers. I have no real objection to changing though; lmk if you'd prefer that.

Reply:
Done. (I actually don't know how many people are familiar with `:any-link` but it's not hard to learn if the reader isn't, so it's fine.)